Specs/Core/Cartesian4Spec.js

```diff
@@ -1084,20 +1084,21 @@ describe("Core/Cartesian4", function () {
   });
 
   it("packs and unpacks floating point values for representation as uint8 4-vectors", function () {
-    var float = 123.456;
-    var packedFloat = Cartesian4.packFloat(float);
-    expect(0 <= packedFloat.x && packedFloat.x <= 255).toBe(true);
-    expect(0 <= packedFloat.y && packedFloat.y <= 255).toBe(true);
-    expect(0 <= packedFloat.z && packedFloat.z <= 255).toBe(true);
-    expect(0 <= packedFloat.w && packedFloat.w <= 255).toBe(true);
-
-    var unpackedFloat = Cartesian4.unpackFloat(packedFloat);
-    expect(
-      CesiumMath.equalsEpsilon(float, unpackedFloat, CesiumMath.EPSILON7)
-    ).toBe(true);
-
-    var packedZero = Cartesian4.packFloat(0);
-    expect(packedZero).toEqual(Cartesian4.ZERO);
+    function testFloat(float) {
+      var packedFloat = Cartesian4.packFloat(float);
+      expect(0 <= packedFloat.x && packedFloat.x <= 255).toBe(true);
+      expect(0 <= packedFloat.y && packedFloat.y <= 255).toBe(true);
+      expect(0 <= packedFloat.z && packedFloat.z <= 255).toBe(true);
+      expect(0 <= packedFloat.w && packedFloat.w <= 255).toBe(true);
+
+      var unpackedFloat = Cartesian4.unpackFloat(packedFloat);
+      expect(unpackedFloat).toEqual(float);
+    }
+    testFloat(0.0);
+    testFloat(-1.0);
+    testFloat(+1.0);
+    testFloat(123.5);
+    testFloat(16777216);
```

Comment:
I updated the test:
```Javascript
   it("packs and unpacks floating point values for representation as uint8 4-vectors", function () {
    function testFloat(float) {
      var packedFloat = Cartesian4.packFloat(float);
      expect(0 <= packedFloat.x && packedFloat.x <= 255).toBe(true);
      expect(0 <= packedFloat.y && packedFloat.y <= 255).toBe(true);
      expect(0 <= packedFloat.z && packedFloat.z <= 255).toBe(true);
      expect(0 <= packedFloat.w && packedFloat.w <= 255).toBe(true);

      var unpackedFloat = Cartesian4.unpackFloat(packedFloat);
      expect(unpackedFloat).toEqual(float);
    }

    function testFloatNaN(float) {
      expect(float).toBeNaN();
      var packedFloat = Cartesian4.packFloat(float);
      var unpackedFloat = Cartesian4.unpackFloat(packedFloat);
      expect(unpackedFloat).toBeNaN();
    }

    testFloat(0.0);
    testFloat(-1.0);
    testFloat(+1.0);
    testFloat(123.5);
    testFloat(16777216);

    testFloat(Infinity) // 64bit infinity -> 32bit infinity
    testFloat(-Infinity) // 64bit infinity -> 32bit infinity
    testFloatNaN(NaN) // 64bit nan -> 32bit nan

    const special = Float32Array.of(Infinity, -Infinity, NaN);
    testFloat(special[0])
    testFloat(special[1])
    testFloatNaN(special[2])
    testFloat(Number.MAX_VALUE) // fails
  });
  ```
The function should probably detect values that cannot fit inside a 32-bit float and trigger an exception

Reply:
@Samulus good catch. Fixed